lego/apps/events/models.py

```diff
@@ -84,6 +84,20 @@ def admin_register(self, user, pool, admin_reason, feedback=''):
         :param feedback: Feedback to organizers
         :return: The registration
         """
+        if not pool:
```

Comment:
Can you write this without the duplicate transaction and update_or_create? `if not pool or self.pools.filter(id=pool.id):` and handle the locked pool counter in another if pool statement?